File: qodana/core/test-data/PhpCoverageInspectionTest/sources/src/FooCls.php
<?php

namespace qodana;

class FooCls
{
    public function foo()
    {
        return 42;
    }

    public function bar($i)
    {
        if ($i % 2) {
            fun2();
            return 42;
        }
        return -1;
    }

    public function baz($i)
    {
        if ($i % 2) {
            return 42;
        }
        $i++;
        $i++;
        return -1;
    }

    public function foobar()
    {
        return 42;
    }
}

function fun1()
{
    phpinfo();
}

function fun2()
{
    return 42;
}

new class {
    public function foo()
    {
        function ($number) {
            return $number * 2;
        };

        function ($string) {
            return strtoupper($string);
        };

        function ($array) {
            return array_reverse($array);
        };
    }
};

$fn = fn($x = 42) =>
            /*foo*/
            $x > 15
            ? 1
            : 42;

class BaseClass {
    function __construct() {
        print "In BaseClass constructor\n";
    }
}

File: qodana/core/test-data/PyCoverageInspectionTest/sources/src/FooCls.py
class FooCls:
    def foo(self):
        if 1 == 1:
            return 42
        ret = 10
        ret += 12
        ret -= 11

        return ret ** 2

    def bar(self, i):
        if i % 2 == 1:
            fun2()
            return 42
        fun1()
        self.baz(i)

        return -1

    def baz(self, i):
        if i % 2 == 1:
            return 42
        elif i == 3:
            return -2
        elif i == 5:
            return -2
        return -1

    def foobar(self):
        return 42


def fun1():
    ret = "A"
    ret += "b"
    return ret

def fun2():
    return 42

class BaseClass:
    def __init__(self):
        print("In base class constructor")
        print("In base class constructor 2")
